test/optim/test_initializers.py

```diff
@@ -187,64 +221,152 @@ def test_gen_batch_initial_conditions(self):
         bounds = torch.stack([torch.zeros(2), torch.ones(2)])
         mock_acqf = MockAcquisitionFunction()
         mock_acqf.objective = lambda y: y.squeeze(-1)
-        for dtype in (torch.float, torch.double):
+        for (
+            dtype,
+            nonnegative,
+            seed,
+            init_batch_limit,
+            ffs,
+            sample_around_best,
+        ) in product(
+            (torch.float, torch.double),
+            [True, False],
+            [None, 1234],
+            [None, 1],
+            [None, {0: 0.5}],
+            [True, False],
+        ):
```

Comment:
This runs 64 combinations of tests, which is a bit too much. It'd be a lot more efficient to just run a handful of combinations that cover all options.
```suggestion
        ) in (
            (torch.float, True, None, None, None, True),
            (torch.double, False, 1234, 1, {0: 0.5}, False),
            (torch.double, True, 1234, None, {0: 0.5}, True),  # could add a few more that are interesting
        ):
```

Reply:
will do, here the change was just combining the two loops based on a comment from @Balandat to remove an indent.